crates/virtio-bindings/src/virtio_net.rs

```diff
@@ -536,17 +607,308 @@ fn bindgen_test_layout_virtio_net_config() {
         );
     }
     test_field_duplex();
+    fn test_field_rss_max_key_size() {
+        assert_eq!(
+            unsafe {
+                let uninit = ::std::mem::MaybeUninit::<virtio_net_config>::uninit();
+                let ptr = uninit.as_ptr();
+                ::std::ptr::addr_of!((*ptr).rss_max_key_size) as usize - ptr as usize
+            },
+            17usize,
+            concat!(
+                "Offset of field: ",
+                stringify!(virtio_net_config),
+                "::",
+                stringify!(rss_max_key_size)
+            )
+        );
+    }
+    test_field_rss_max_key_size();
+    fn test_field_rss_max_indirection_table_length() {
+        assert_eq!(
+            unsafe {
+                let uninit = ::std::mem::MaybeUninit::<virtio_net_config>::uninit();
+                let ptr = uninit.as_ptr();
+                ::std::ptr::addr_of!((*ptr).rss_max_indirection_table_length) as usize
+                    - ptr as usize
+            },
+            18usize,
+            concat!(
+                "Offset of field: ",
+                stringify!(virtio_net_config),
+                "::",
+                stringify!(rss_max_indirection_table_length)
+            )
+        );
+    }
+    test_field_rss_max_indirection_table_length();
+    fn test_field_supported_hash_types() {
+        assert_eq!(
+            unsafe {
+                let uninit = ::std::mem::MaybeUninit::<virtio_net_config>::uninit();
+                let ptr = uninit.as_ptr();
+                ::std::ptr::addr_of!((*ptr).supported_hash_types) as usize - ptr as usize
+            },
+            20usize,
+            concat!(
+                "Offset of field: ",
+                stringify!(virtio_net_config),
+                "::",
+                stringify!(supported_hash_types)
+            )
+        );
+    }
+    test_field_supported_hash_types();
 }
 #[repr(C)]
-#[derive(Debug, Default, Copy, Clone, PartialEq)]
+#[derive(Copy, Clone)]
 pub struct virtio_net_hdr_v1 {
     pub flags: __u8,
     pub gso_type: __u8,
     pub hdr_len: __virtio16,
     pub gso_size: __virtio16,
+    pub __bindgen_anon_1: virtio_net_hdr_v1__bindgen_ty_1,
+    pub num_buffers: __virtio16,
+}
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub union virtio_net_hdr_v1__bindgen_ty_1 {
+    pub __bindgen_anon_1: virtio_net_hdr_v1__bindgen_ty_1__bindgen_ty_1,
+    pub csum: virtio_net_hdr_v1__bindgen_ty_1__bindgen_ty_2,
+    pub rsc: virtio_net_hdr_v1__bindgen_ty_1__bindgen_ty_3,
+}
```

Comment:
@alyssais I think it's not acceptable to lose the traits implementations (Debug, Default and PartialEq) since this might introduce some regressions for consumers of this crate. The only alternative I can think of is by manually implementing these traits for `virtio_net_hdr_v1__bindgen_ty_1`.

Reply:
Well the point about added members in structures doesn't really apply given we already agreed to follow what the kernel does, meaning it's acceptable to have new fields in a structure, and things like JSON encoding/decoding should be able to handle such situation from the consumer's code.
But in general I agree let's not make things too complex to maintain, meaning the second approach using minor (or major) version to introduce breaking changes (from a Rust perspective) would be the way to go IMO.

Reply:
/cc @andreeaflorescu @jiangliu @slp wdyt?

Reply:
If I remember correctly PartialEq, Debug and Default were not added to all structures by bindgen because of an issue in how they're generating the bindings. We added them manually for older versions because otherwise the structures are pretty hard to use. I think we need to find a way to still have these traits implemented. Let me do some digging and figure out how we handle this in the past.

Reply:
Ah, that explains why I couldn't find a version of bindgen that exactly reproduced the bindings as they were before, so I suppose the generated bindings files were just edited?  It would be good to find a way around that, as it's useful for auditability if the generated files can be exactly reproduced.

Reply:
The way that we did this is to have one commit with the autogenerated bindings, and then separate commits for the changes. Than the changes were also added to the repository as patch files so that the process can be reproduced.

Reply:
If you have some more time to invest in this, we can also find a better way of generating them while still keeping the derives that we need. In any case, I don't think it's acceptable to not have them.

Reply:
@alyssais did you give this more thought? Are you still interested in regenerating the bindings?

Reply:
Hi! I'm still interested, but likely won't get a chance to look at it until I'm back from vacation in mid-October.

Reply:
Ok, no problem, thanks for letting me know!

Reply:
Quick update: after I came back from my vacation I immediately got a bad flu, so I'm just catching up now.  I'll look at this soon. :)